pkg/collector/corechecks/cluster/topologycollectors/service_collector.go

```diff
@@ -212,6 +227,65 @@ func (sc *ServiceCollector) serviceToStackStateComponent(service v1.Service) *to
 	return component
 }
 
+func (sc *ServiceCollector) serviceToExternalServiceComponent(service v1.Service) *topology.Component {
+	log.Tracef("Mapping kubernetes pod ExternalName service to extra StackState component: %s", service.String())
+	// create identifier list to merge with StackState components
+	identifiers := make([]string, 0)
+
+	if service.Spec.ExternalName != "None" && service.Spec.ExternalName != "" {
+		identifiers = append(identifiers, fmt.Sprintf("urn:endpoint:/%s", service.Spec.ExternalName))
+		// If targetPorts are specified, use those
+		for _, port := range service.Spec.Ports {
+			// map all the node ports
+			if port.Port != 0 {
+				identifiers = append(identifiers, fmt.Sprintf("urn:endpoint:/%s:%s:%d", sc.GetInstance().URL, service.Spec.ExternalName, port.Port))
+			}
+		}
+
+		addrs, err := sc.DNS(service.Spec.ExternalName)
```

Comment:
What is the use case for checking whether this can be reached or not? I imagine there is functionality in Kubernetes that validates whether this endpoint is reachable by the pod and produces an error if not. If not we have a nice use case here to inform users of incorrect configuration. 

So I would make the case that we always want to map these identifiers seeing as they are configured in Kubernetes as the "endpoint". The "reach-ability" of the endpoint is a health state in my opinion.

I'm also concerned that we might potentially be making a decent amount of requests in a large cluster, for something that is not necessarily beneficial to the user.

Reply:
We're not doing reachability, but DNS resolution. As the system typically caches DNS results, this should not be a problem. We need to know the IP addresses in order to be able to detect traffic flow using the tracing agent.